tests/cases/fourslash/quickInfoInJsdocInTsFile1.ts

```diff
@@ -0,0 +1,62 @@
+/// <reference path='fourslash.ts' />
+
+//// /** @type {() => { /*1*/data: string[] }} */
+//// function test(): { data: string[] } {
+////   return {
+////     data: [],
+////   };
+//// }
+////
+//// /** @returns {{ /*2*/data: string[] }} */
+//// function test2(): { data: string[] } {
+////   return {
+////     data: [],
+////   };
+//// }
+////
+//// /** @type {{ /*3*/bar: string; }} */
+//// const test3 = { bar: '' };
+////
+//// type SomeObj = { bar: string; };
+//// /** @type {SomeObj/*4*/} */
```

Comment:
In TS files this is just ignored, in the sense that it doesn't impact types at all. So technically we could just avoid displaying anything here. I concluded though that it would actually be less useful and that it would be confusing - usually the user expects some feedback in form of a tooltip and not returning any info here could feel like something is off, like it's broken.

Since this isn't used for type-checking there is no guarantee that it "matches" the declaration below - I think it's an acceptable tradeoff. If somebody is writing JSDoc types in TS files then it's their responsibility to make them match.

Reply:
cc @jakebailey since we were talking about this a few hours ago